src/operators/bufferCount.ts

```diff
@@ -64,7 +64,10 @@ class BufferCountSubscriber<T> extends Subscriber<T> {
     const destination = this.destination;
     const buffers = this.buffers;
     while (buffers.length > 0) {
-      destination.next(buffers.shift());
+      var buffer = buffers.shift();
+      if (buffer.length > 0) {
```

Comment:
This check is to prevent bufferCount emits empty buffer element at the end of observable on certain condition, such as ```Observable.range(0,6).bufferCount(2) ``` where empty buffer is pushed then no element is actually counted in.